CarAccidentErrorHandling/CarAccidentErrorHandling.ts

```diff
@@ -0,0 +1,118 @@
+class SlipperyRoadAccidentException extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = this.constructor.name;
+  }
+}
+
+class CollisionAccidentException extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = this.constructor.name;
+  }
+}
+
+class HitAndRunAccidentException extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = this.constructor.name;
+  }
+}
+
+class EnvironmentalAccidentException extends Error {
+  constructor(message: string) {
+    super(message);
+    this.name = this.constructor.name;
+  }
+}
+
+function driveSafely() {
+  const isSlipperyRoad = true;
```

Comment:
Cannot we use enums here?

Reply:
Updated the code with more optimal approach.